webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAIPromptService.java

```diff
@@ -0,0 +1,169 @@
+package com.box.l10n.mojito.service.ai.openai;
+
+import com.box.l10n.mojito.JSR310Migration;
+import com.box.l10n.mojito.entity.AIPrompt;
+import com.box.l10n.mojito.entity.AIPromptContextMessage;
+import com.box.l10n.mojito.entity.AIPromptType;
+import com.box.l10n.mojito.entity.PromptType;
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.entity.RepositoryAIPrompt;
+import com.box.l10n.mojito.rest.ai.AIException;
+import com.box.l10n.mojito.rest.ai.AIPromptContextMessageCreateRequest;
+import com.box.l10n.mojito.rest.ai.AIPromptCreateRequest;
+import com.box.l10n.mojito.service.ai.PromptService;
+import com.box.l10n.mojito.service.repository.RepositoryRepository;
+import io.micrometer.core.annotation.Timed;
+import jakarta.transaction.Transactional;
+import java.time.ZonedDateTime;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+
+@Service
+@ConditionalOnProperty(value = "l10n.ai.service.type", havingValue = "OpenAI")
+public class OpenAIPromptService implements PromptService {
```

Comment:
Could this PromptService be used for any backend LLM engine or is it OpenAI specific?

From what I can tell - the main reason why it's marked as OpenAI is due to the PromptContext being of type OpenAIPromptContextMessageType - that lists a set of "known" roles.
Don't have a strong opinion on it - but seems to be a good enough PromptService to be the default for everything that would be stored - no matter who would actually execute the prompts.

Reply:
I guess it could be used for any LLM engine with the caveat that its been developed against an OpenAI backend so may be biased towards that interface.

There's nothing OpenAI specific in it though so happy to generalise it